zio-http/src/test/scala/zio/http/QueryParamsSpec.scala

```diff
@@ -0,0 +1,220 @@
+package zio.http
+
+import zio.test.Assertion.equalTo
+import zio.test.{ZIOSpecDefault, _}
+
+object QueryParamsSpec extends ZIOSpecDefault {
+
+  def spec =
+    suite("QueryParams")(
+      suite("-")(
+        test("removes query param") {
+          val gens = Gen.fromIterable(
```

Comment:
We can use a Gen with random alpha numerics and check for symmetry between encoding and decoding